File: packages/typescript/src/resources/pricing.ts
/**
 * The `pricing` resource: your workspace price list.
 * @module
 */
import type { PriceList, PricingParams, RequestOptions } from '../models.js';
import { Resource, wireQuery } from './base.js';

/** Accessed as `client.pricing`. */
export class Pricing extends Resource {
  get(params: PricingParams = {}, options?: RequestOptions): Promise<PriceList> {
    return this.one({ method: 'GET', path: '/v1/pricing', query: wireQuery(params) }, options);
  }
}

File: packages/rust/src/resources/pricing.rs
//! The `pricing` resource.

use crate::error::Result;
use crate::http::HttpClient;
use crate::models::{PriceList, PricingParams};
use crate::util::query;

/// Accessed as `client.pricing()`.
#[derive(Debug, Clone)]
pub struct Pricing {
    http: HttpClient,
}

impl Pricing {
    pub(crate) fn new(http: HttpClient) -> Self {
        Self { http }
    }

    /// The workspace price list for a product and optional country.
    pub async fn get(&self, params: PricingParams) -> Result<PriceList> {
        let q = query(&[("product", params.product), ("country", params.country)]);
        self.http.get(&format!("/v1/pricing{q}")).await
    }
}
